Mathlib/Topology/Compactness/HilbertCubeEmbedding.lean

```diff
@@ -0,0 +1,64 @@
+/-
+Copyright (c) 2025 Vasilii Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasilii Nesterov
+-/
+import Mathlib.Topology.UnitInterval
+
+/-!
+# Every compact metric space can be embedded into the Hilbert cube.
+
+In this file we prove `exists_closed_embedding_to_hilbert`: every compact metric space can be
+embedded into the Hilbert cube (`ℕ → unitInterval`).
+-/
+
+open Classical in
+/-- Every compact metric space can be embedded into the Hilbert cube. -/
+theorem exists_closed_embedding_to_hilbert_cube (X : Type*) [MetricSpace X] [CompactSpace X] :
+    ∃ f : X → (ℕ → unitInterval), Topology.IsClosedEmbedding f := by
+  let f : X → (ℕ → unitInterval) :=
+    if h : Nonempty X then
+      let s := TopologicalSpace.denseSeq X
+      fun x i =>
+        let d := dist x (s i)
+        let diam := Metric.diam (Set.univ : Set X)
+        have hd1 : d ≤ diam := by
+          simp [diam]
+          apply Metric.dist_le_diam_of_mem _ (by simp) (by simp)
+          rwa [← Metric.compactSpace_iff_isBounded_univ]
+        have hd2 : (d / diam) ∈ unitInterval := by
+          simp only [Set.mem_Icc, diam]
+          exact ⟨by positivity, div_le_one_of_le₀ hd1 (by positivity)⟩
+        ⟨_, hd2⟩
+    else
+      fun x i ↦ 0
```

Comment:
I think you can avoid the if-then-else: start by introducing x, and then you can build your `Nonempty` instance using `x` to build the rest.

Reply:
Okay, I see the problem: if you do so, then Lean thinks the dense sequence depends on `x`, so proving continuity is a mess. You can ignore my comment then.